### ISSUE TITLE
175-vscode-cmake-qt

### ISSUE THREAD

**Issue by retifrav** [retifrav/retifrav.github.io#20]
[Visual Studio Code and CMake for Qt](https://retifrav.github.io/blog/2019/05/11/vscode-cmake-qt/) comments.

**Comment by studoot** [comment]
`find_package` and `target_link_libraries` come into their own when you have multiple targets. `find_packages` locates the Qt libraries, while `target_link_libraries` allows you to 

For example, if I had a library (called utils) that doesn't use Qt at all, another library (called model) that uses Qt Core, and an executable (called app) that uses Qt's GUI components, I could do something like this:

```cmake
project(my-code)

find_package(Qt5 REQUIRED COMPONENTS Widgets Core)

add_library(utils <sources>)

add_library(model <sources>)
target_link_libraries(model utils Qt5::Core)

add_executable(app <sources>)
target_link_libraries(app model Qt5::Widgets)
```

I've been using CMake for a while and have found it pretty good, even for the custom things - adding things like test coverage, use of sanitisers, static analysis etc is all pretty simple, and can be moved into custom modules so you can reuse the same CMake code on multiple projects.

I've also found CMake to be the easiest tool for building C++ projects that are required for multiple platforms (mainly Linux & Windows for me).

**Comment by retifrav** [comment]
Indeed, that makes sense. It didn’t occur to me that I can have several targets with a different set of libraries to link with.

**Comment by KapitanPL** [comment]
How do you run it? 
I got so far, but when trying to run the binary it complains that it can not find Qt dlls. Is it necessary to add some deploy step?

**Comment by retifrav** [comment]
If Qt is properly registered in your system, then `find_package(Qt5...` should take care of it. However, as I recall, if path to Qt libraries is not in your `PATH`, then you should either add it there or just copy required libraries alongside your executable.
